static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,122 @@
+            reqwest({
+                url: '/email-prefs',
+                method: 'POST',
+                data: formQueryString,
+                error: function () {
+                    renderErrorMessage(buttonEl);
+                },
+                success: function (response) {
+                    var subscriptionState;
+                    try {
+                        // Why array.length rather than typeof; there are multiple circumstances where the value could be undefined:
+                        // nothing came back at all
+                        // stuff came back but that one field is undefined
+                        // the data came back in the format expected as above, but with the subscribeTo value undefined
+                        if (response.subscriptions.length < 1) {
+                            subscriptionState = false;
+                        } else {
+                            subscriptionState = response.subscriptions[0].subscribedTo;
+                        }
+                    } catch (err) {
+                        renderErrorMessage(buttonEl);
+                    } finally {
+                        updateButton(buttonEl, subscriptionState);
+                    }
+                }
+            });
+        });
+    }
+
+    function enhanceEmailPreferences() {
+        $.forEachElement('.email-subscription__button', useReqwest);
+    }
+
+    function encodeFormData(csrfToken, buttonVal, htmlPreference) {
+        return 'csrfToken=' + encodeURIComponent(csrfToken) + '&' +
+        'addEmailSubscription=' + encodeURIComponent(buttonVal) + '&' +
+        'htmlPreference=' + encodeURIComponent(htmlPreference);
+    }
+
+    function renderErrorMessage(buttonEl) {
+        // appends an error message on the parent div of the button
+        var errorMessage = 'Sorry, an error has occurred, please refresh the page and try again';
+        return fastdom.write(function () {
+            var insertionPoint = $.ancestor(buttonEl, 'email-subscription');
+            // Only append an error message once for each email subscription DIV
+            if (qwery('.form__error', insertionPoint).length < 1) {
+                var errorMessageDiv = document.createElement('div');
+                errorMessageDiv.innerHTML = errorMessage;
+                errorMessageDiv.classList.add('form__error');
+                insertionPoint.appendChild(errorMessageDiv);
+            }
+        });
+    }
+
+    function updateButton(buttonEl, subscriptionState) {
+        // do this after successful POST request to update the appearance and value of the button
+        var buttonVal = buttonEl.value;
+        if (subscriptionState === true) {
+            fastdom.write(function () {
+                buttonEl.value = 'unsubscribe-' + buttonVal;
+                buttonEl.innerHTML = 'Unsubscribe';
+                $.ancestor(buttonEl, 'email-subscription').classList.add('email-subscription--subscribed');
```

Comment:
`classList` won't work in IE9, and I *guess* we do want to aim for compatibility on this page since it's fundamental functionality. use bonzo's `addClass()`